src/bin/lfortran.cpp

```diff
@@ -1714,22 +1682,33 @@ int link_executable(const std::vector<std::string> &infiles,
             std::cout << "The command '" + cmd + "' failed." << std::endl;
             return 10;
         }
-        if (compiler_options.run) {
-            std::string run_cmd = (t == "x86_64-pc-windows-msvc") ? outfile : "./" + outfile;
-            int err = system(run_cmd.c_str());
-            if (err != 0) {
-                if (0 < err && err < 256) {
-                    return err;
-                } else {
-                    return 1;
-                }
-            }
-        }
-        return 0;
     } else {
         LCOMPILERS_ASSERT(false);
         return 1;
     }
+
+    if (outfile != file_name + ".out" || compiler_options.arg_o != "") {
```

Comment:
This is very rare case, but I think this might fail for the case `lfortran x.f90 -o x.out`, so let's only do?
```suggestion
    if ( compiler_options.arg_o != "" ) {
```